gradoop-flink/src/main/java/org/gradoop/flink/model/impl/operators/tpgm/aggregation/AverageDuration.java

```diff
@@ -0,0 +1,94 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradoop.flink.model.impl.operators.tpgm.aggregation;
+
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.gradoop.common.model.impl.pojo.temporal.TemporalElement;
+import org.gradoop.common.model.impl.properties.PropertyValue;
+import org.gradoop.flink.model.api.tpgm.functions.TemporalAttribute;
+import org.gradoop.flink.model.api.tpgm.functions.aggregation.TemporalAggregateFunction;
+import org.gradoop.flink.model.impl.operators.aggregation.functions.BaseAggregateFunction;
+import org.gradoop.flink.model.impl.operators.aggregation.functions.average.Average;
+
+import java.util.Arrays;
+import java.util.Objects;
+
+/**
+ * Calculate the average duration of a time interval of temporal elements.
+ * Time intervals can be {@link TemporalAttribute}s.
+ * Time intervals with either the start or end time set to the respective default value will be
+ * ignored.
+ */
+public class AverageDuration extends BaseAggregateFunction
+  implements Average, TemporalAggregateFunction {
+
+  /**
+   * A property value containing the number {@code 1}, as a {@code long}.
+   */
+  private static final PropertyValue ONE = PropertyValue.create(1L);
+
+  /**
+   * Selects which time-interval is considered by this aggregate function.
+   */
+  private final TemporalAttribute interval;
+
+  /**
+   * Create an instance of the {@link AverageDuration} aggregate function.
+   *
+   * @param aggregatePropertyKey The aggregate property key.
+   * @param interval             The time-interval to consider.
+   */
+  public AverageDuration(String aggregatePropertyKey, TemporalAttribute interval) {
+    super(aggregatePropertyKey);
+    this.interval = Objects.requireNonNull(interval);
+  }
+
+  /**
+   * Get the duration of a time interval as the aggregate value from a temporal element.
+   * The duration will be returned in a format used by the {@link Average} aggregation.
+   * The increment will be ignored, if the start of the end time of the time interval is set
+   * to a default value.
+   *
+   * @param element The temporal element.
+   * @return The duration of the time interval, in the internal .
+   */
+  @Override
+  public PropertyValue getIncrement(TemporalElement element) {
+    Tuple2<Long, Long> timeInterval;
+    switch (interval) {
+    case TRANSACTION_TIME:
+      timeInterval = element.getTransactionTime();
+      break;
+    case VALID_TIME:
+      timeInterval = element.getValidTime();
+      break;
+    default:
+      throw new IllegalArgumentException(
+        "Temporal attribute " + interval + " is not supported " + "by this aggregate function.");
+    }
+    if (timeInterval.f0 == null || timeInterval.f0.equals(TemporalElement.DEFAULT_TIME_FROM) ||
```

Comment:
Change the logic here: If transaction time is chosen, we will not check for default elements. Only for valid times.

Reply:
You UnitTests has to be refactored after this.

Reply:
This is outdated.